workflows/sweep_simulate.snake

```diff
@@ -0,0 +1,324 @@
+        for left, right in windows
+        for seed in seed_array
+        for dfe in dfe_list
+        for annot in annotation_list
+        for coeff in sel_coeffs
+        for time in times
+        for suffix in [".trees", ".stats.npz"] 
+        ]
+
+print(windows.shape)
+
+rule boundary_sims:
+    input:
+    output:
+        output_dir + "/simulated_data/sweeps/boundary_sims/sim_{seed}_{region_size}.trees"
+    resources: time=6000, mem_mb=100000
+    run:
+        model = species.get_demographic_model(demo_model["id"])
+        mut_rate = model.mutation_rate 
+        # Use new sample specification, give selected pop and sample size as config arguments
+        samples = demo_model["samples"]
+        dfe = species.get_dfe(dfe_list[0])
+        # reducing number of del muts bc it's too much!
+        dfe.proportions = [1-(0.7*0.25), 0.7*0.25]
+        clen = int(wildcards.region_size)
+        contig = species.get_contig(length=clen)
+        contig.add_dfe(intervals=[[0, clen]], DFE=dfe)# applying DFE over some equally spaced 
+        contig.mutation_rate = mut_rate
+        engine = stdpopsim.get_engine("slim")
+        ts = engine.simulate(
+            model,
+            contig,
+            samples,
+            slim_scaling_factor=config["slim_scaling_factor"],
+            slim_burn_in=config["slim_burn_in"],
+            seed=int(wildcards.seed),
+        )
+        ts.dump(output[0])
+
+rule neutral:
+    input:
+    output:
+        output_dir + f"/simulated_data/sweeps/neutral/{{seed}}/sim_{chrom}_{{left}}_{{right}}.trees", 
+    resources: time=3000, mem_mb=2000
+    run:
+        model = species.get_demographic_model(demo_model["id"])
+        mutation_rate = model.mutation_rate
+        extended_events = None
+        # Use new sample specification, give selected pop and sample size as config arguments
+        samples = demo_model["samples"]
+
+        # Extract contig spanning window of chromosome
+        window_left = int(wildcards.left)
+        window_right = int(wildcards.right)
+        contig = species.get_contig(chrom, genetic_map=genetic_map_id, left=window_left, right=window_right)
+
+        contig.mutation_rate = mutation_rate
+        engine = stdpopsim.get_engine("slim")
+        ts = engine.simulate(
+            model,
+            contig,
+            samples,
+            extended_events=extended_events,
+            slim_scaling_factor=config["slim_scaling_factor"],
+            slim_burn_in=config["slim_burn_in"],
+            seed=int(wildcards.seed),
+        )
+        ts = add_window_metadata(ts, window_left, window_right, window_left, window_right)
+        ts.dump(output[0])
+
+rule bgs:
+    input:
+    output:
+        output_dir + f"/simulated_data/sweeps/bgs/{{annot}}/{{dfe}}/{{seed}}/sim_{chrom}_{{left}}_{{right}}.trees", 
+    resources: time=3000, mem_mb=4000
```

Comment:
these resources seem specific -- was this tuned?

Reply:
the memory was tuned a bit, the time was not.